### PR TITLE
fix #8386 chore(nimbus): Add ruff and circle ci check for cirrus ✅ 

### LINKED ISSUE #8386: Set up Ruff Linter
We use ruff currently in the experimenter, we will use the same for the cirrus service. [Ruff](https://pypi.org/project/ruff/0.0.89/)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3215)

### PR DESCRIPTION
**Pull request by yashikakhurana**
Because

- We want to add a linter on the Cirrus service and wanted to have CI runs the cirrus check

This commit

- Add [Ruff linter](https://pypi.org/project/ruff/) 💪 
- New circle ci job for `cirrus_check`
- Cirrus check - runs the test case and ruff linting
- New make command for check `make cirrus_check`
- New make command to fix  formatting issues using `make cirrus_code_format`
- Circle ci cirrus check will only run if the changes are in `/cirrus` directory
- Experimenter check will only run if the changes are in `/experimenter` directory
- Remove requirements.txt and uses poetry only
closes #8386 #8388

### CODE REVIEW

**Comment by yashikakhurana** [comment]
@jrbenny35 is it safe to avoid an integration test like remote settings, enrollments test, etc on CI if there are no changes in `/experimenter` directory? If yes I can file a separate ticket for that, just wanted to double-check, thanks

**Comment by yashikakhurana** [comment]
okay so finally @jaredlockhart @jeddai @eliserichards @jrbenny35 this is ready for review 🙏

**Comment by jaredlockhart** [comment]
> @jrbenny35 is it safe to avoid an integration test like remote settings, enrollments test, etc on CI if there are no changes in `/experimenter` directory? If yes I can file a separate ticket for that, just wanted to double-check, thanks

Yeah that's an interesting question, I'd like to hear what everybody thinks, let's add it to the next Nimbus engineering meeting and we can talk about it together.

File: cirrus/server/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,29 @@
+# Use official Python runtime as a parent image
 FROM python:3.11.2-slim-buster
+
+# Set working directory for the container
 WORKDIR /usr/src/cirrus
```
Comment (jaredlockhart) [inline cirrus/server/Dockerfile:5]: Oh yeah we're gonna wanna make this into `/app/` to stay consistent with other service containers.

**Review by jaredlockhart — APPROVED** [review]
Yep generally all looks right to me, couple points.  You could address feedback in this pr before landing or file as followups, whichever works for you.  Looking good @yashika-khurana !

File: .circleci/config.yml [hunk 2]
```diff
@@ -19,8 +19,15 @@ jobs:
       - run:
           name: Run tests and linting
           command: |
-            cp .env.sample .env
-            make check
+            if git diff --name-only main HEAD | grep -E 'experimenter/'
```
[the remark this answers is not in the document]
Comment (yashikakhurana) [reply .circleci/config.yml:22]: yes right here is ticket for that- https://github.com/mozilla/experimenter/issues/8510

File: cirrus/server/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+description = ""
+authors = [""]
+
+[tool.poetry.dependencies]
+python = "^3.10.10"
+ruff = "^0.0.254"
+fastapi = "^0.92.0"
+uvicorn = "^0.20.0"
+pytest = "^7.2.2"
+httpx = "^0.23.3"
```
[the remark this answers is not in the document]
Comment (yashikakhurana) [reply cirrus/server/pyproject.toml:13]: so we are using `TestClient` and under the hood it is based on `httpx`, more [info](https://fastapi.tiangolo.com/advanced/async-tests/)

File: cirrus/server/Dockerfile (hunk 1 above)
Comment (yashikakhurana) [reply cirrus/server/Dockerfile]: good call, thank you :)

**Pull request merged**